applications/OptimizationApplication/tests/control/shape/test_vm_shape_control.py

```diff
@@ -0,0 +1,195 @@
+            cls.optimization_problem.AddComponent(cls.implicit_solid_shape_control)
+            ReadModelPart("solid",cls.solid_implicit)
+            cls.implicit_solid_shape_control.Initialize()
+
+
+            cls.solid_bulk_surf_implicit = cls.model.CreateModelPart("solid_bulk_surf_implicit")
+            cls.solid_bulk_surf_implicit.ProcessInfo[Kratos.DOMAIN_SIZE] = 3
+            solid_parameters = Kratos.Parameters("""{
+                "controlled_model_part_names": ["solid_bulk_surf_implicit.design"],
+                "filter_settings": {
+                    "type" : "bulk_surface_implicit",
+                    "radius": 0.5
+                },
+                "fixed_model_parts" : {
+                    "solid_bulk_surf_implicit.fixed" : [true,true,true]
+                }
+            }""")
+
+            cls.implicit_bulk_surf_shape_control = VertexMorphingShapeControl("solid_bulk_surf_implicit", cls.model, solid_parameters, cls.optimization_problem)
+            cls.optimization_problem.AddComponent(cls.implicit_bulk_surf_shape_control)
+            ReadModelPart("solid",cls.solid_bulk_surf_implicit)
+            cls.implicit_bulk_surf_shape_control.Initialize()
+
+    @classmethod
+    def tearDownClass(cls):
+        with kratos_unittest.WorkFolderScope(".", __file__):
+            DeleteFileIfExisting("shell.time")
+
+    def test_GetControlField(self):
+        implicit_control_field = self.implicit_shape_control.GetControlField()
+        self.assertAlmostEqual(KratosOA.ExpressionUtils.NormInf(implicit_control_field), 1.0903407171261381, 10)
+        explicit_control_field = self.explicit_shape_control.GetControlField()
+        self.assertAlmostEqual(KratosOA.ExpressionUtils.NormInf(explicit_control_field), 0.0000, 4)
+        solid_explicit_control_field = self.explicit_solid_shape_control.GetControlField()
+        self.assertAlmostEqual(KratosOA.ExpressionUtils.NormInf(solid_explicit_control_field), 0.0000, 4)
+        solid_implicit_control_field = self.implicit_solid_shape_control.GetControlField()
+        self.assertAlmostEqual(KratosOA.ExpressionUtils.NormInf(solid_implicit_control_field), 9.0000, 4)
+        solid_implicit_bulk_surf_control_field = self.implicit_bulk_surf_shape_control.GetControlField()
+        self.assertAlmostEqual(KratosOA.ExpressionUtils.NormInf(solid_implicit_bulk_surf_control_field), 63.476190476190496, 4)
+
+
+    def test_GetPhysicalField(self):
+        shape_field = self.implicit_shape_control.GePhysicalField()
+        self.assertAlmostEqual(KratosOA.ExpressionUtils.NormL2(shape_field), 20.976176963410882, 10)
+        shape_field = self.explicit_shape_control.GePhysicalField()
+        self.assertAlmostEqual(KratosOA.ExpressionUtils.NormL2(shape_field), 20.976176963410882, 10)
+        shape_field = self.explicit_solid_shape_control.GePhysicalField()
```

Comment:
This shape field includes volume nodes or only surface?

Reply:
In case of bulk_surface_implicit filter it contains all nodes (volume+surface), otherwise only surface nodes